### ISSUE TITLE
Loss becomes zero and grad_norm turns to NaN during stage 2 training

### ISSUE THREAD

**Issue by ChenLLong** [microsoft/GUI-Actor#20]
hello，I'm currently using the official training script  for two-stage training process. Everything works fine during the first stage, but I encountered an issue during the second stage training.

Specifically, after a few steps (or at a certain point), the loss suddenly drops to zero , and the grad_norm becomes NaN , which causes the training to fail. 

base model: qwen2.5-vl-7b

here is the training log：

{'loss': 2.5552, 'grad_norm': 6436024.447018536, 'learning_rate': 4.999937520594028e-06, 'epoch': 0.03}                        
{'loss': 2.7004, 'grad_norm': 2915489.527697916, 'learning_rate': 4.999917946994077e-06, 'epoch': 0.03}                        
{'loss': 2.668, 'grad_norm': 25656094.98940503, 'learning_rate': 4.999895710381286e-06, 'epoch': 0.03}                         
{'loss': 2.623, 'grad_norm': 24828488.256136153, 'learning_rate': 4.9998708107793414e-06, 'epoch': 0.03}                       
{'loss': 2.81, 'grad_norm': 36071025.201348156, 'learning_rate': 4.999843248214766e-06, 'epoch': 0.03}                         
{'loss': 2.9197, 'grad_norm': 6908930.777177996, 'learning_rate': 4.999813022716923e-06, 'epoch': 0.03}                        
{'loss': 3.3198, 'grad_norm': 1759233.4381649764, 'learning_rate': 4.999780134318008e-06, 'epoch': 0.03}                       
{'loss': 3.8148, 'grad_norm': 964284397877.13, 'learning_rate': 4.999744583053056e-06, 'epoch': 0.03}                          
{'loss': 4.4811, 'grad_norm': nan, 'learning_rate': 4.9997063689599365e-06, 'epoch': 0.03}                                     
{'loss': 0.0, 'grad_norm': nan, 'learning_rate': 4.999665492079358e-06, 'epoch': 0.04}                                         
{'loss': 0.0, 'grad_norm': nan, 'learning_rate': 4.999621952454863e-06, 'epoch': 0.04}                                         
{'loss': 0.0, 'grad_norm': nan, 'learning_rate': 4.999575750132833e-06, 'epoch': 0.04}                                         
{'loss': 0.0, 'grad_norm': nan, 'learning_rate': 4.999526885162484e-06, 'epoch': 0.04}                                         
{'loss': 0.0, 'grad_norm': nan, 'learning_rate': 4.999475357595868e-06, 'epoch': 0.04}                                         
{'loss': 0.0, 'grad_norm': nan, 'learning_rate': 4.999421167487876e-06, 'epoch': 0.04}                                         
  4%|██▉                                                                            | 11500/313802 [35:03<102:16:41,  1.22s/it]

Here is my second training script， using learning rate 5e-6：

torchrun --nproc_per_node=8 train.py \
  --deepspeed ./scripts/zero3.json \
  --data_path data/data_config_local.yaml \
  --image_folder "" \
  --model_type ${model_type} \
  --model_name_or_path ${llm_model} \
  --group_by_modality_length True \
  --bf16 True \
  --output_dir ${output_dir} \
  --num_train_epochs 1 \
  --per_device_train_batch_size 1 \
  --per_device_eval_batch_size 4 \
  --gradient_accumulation_steps 1 \
  --eval_strategy "no" \
  --save_strategy "steps" \
  --save_steps 10000 \
  --learning_rate 5e-6 \
  --weight_decay 0. \
  --warmup_ratio 0.03 \
  --lr_scheduler_type "cosine" \
  --logging_steps 100 \
  --tf32 True \
  --model_max_length 24576 \
  --gradient_checkpointing True \
  --dataloader_num_workers 8 \
  --max_pixels 5720064 \
  --unfreeze_all_parameters True \
  --unfreeze_pointer_head False \
  --unfreeze_lm_head False \
  --unfreeze_base_model False \
  --unfreeze_last_n_layers -1 \
  --unfreeze_new_tokens False \
  --unfreeze_visual False \
  --pointer_loss_weight 1.0 \
  --lm_loss_weight 1.0

Does anyone have any idea why this might be happening? Any suggestions or debugging tips would be greatly appreciated!

Thanks for your help!

**Comment by QianhuiWu** [comment]
Hi @ChenLLong, thanks for sharing the problem. We haven’t run into the same issue in our own tests. Could you please share more information about your experiments? For example, how many GPUs did you train on? What data are you using? Have you confirmed that its structure matches the format of the [dataset we released](https://huggingface.co/datasets/cckevinn/GUI-Actor-Data)? Have you tried training with a different base model? If so, how did the loss curves compare?
